# Estadistica Descriptiva

- Unidad de observacion
- Poblacion: todas las unidades de observacion posibles
- Muestra: conjunto de observaciones que contiene las unidades de observacion

### Inferencia Estadistica
- Metodos y tecnicas que permiten inferir en el comportamoento de una poblacion con base en una muestra

In [3]:
import pandas as pd
import numpy as np
#para montar en drive

import os
#print(os.getcwd())
! pip install pingouin
import matplotlib.pyplot as plt
import seaborn as sns
import statsmodels.api as sm
from statsmodels.formula.api import ols
import statsmodels
from scipy import stats
from pingouin import pairwise_ttests #this is for performing the pairwise tests

  Obtaining dependency information for pingouin from https://files.pythonhosted.org/packages/35/2e/8ca90e7edc93bc3d3bdf6daa6d5fc5ae4882994171c3db765365227e1d58/pingouin-0.5.4-py2.py3-none-any.whl.metadata
  Obtaining dependency information for pandas-flavor from https://files.pythonhosted.org/packages/67/1a/bfb5574b215f530c7ac5be684f33d60b299abbebd763c203aa31755f2fb2/pandas_flavor-0.6.0-py3-none-any.whl.metadata
   ---------------------------------------- 0.0/198.9 kB ? eta -:--:--
   -- ------------------------------------- 10.2/198.9 kB ? eta -:--:--
   ---------------------------------------- 198.9/198.9 kB 4.0 MB/s eta 0:00:00


In [4]:
bank = pd.read_csv("C:\\Users\\VICTUS\\Documents\\2024\\DATA SCIENCE\\8. Estadistica Descriptiva\\bank-full (1).csv")
print(bank.shape)
bank.head()

(45211, 17)


age           job  marital  education default  balance housing loan  \
0   58    management  married   tertiary      no     2143     yes   no   
1   44    technician   single  secondary      no       29     yes   no   
2   33  entrepreneur  married  secondary      no        2     yes  yes   
3   47   blue-collar  married    unknown      no     1506     yes   no   
4   33       unknown   single    unknown      no        1      no   no   

   contact  day month  duration  campaign  pdays  previous poutcome   y  
0  unknown    5   may       261         1     -1         0  unknown  no  
1  unknown    5   may       151         1     -1         0  unknown  no  
2  unknown    5   may        76         1     -1         0  unknown  no  
3  unknown    5   may        92         1     -1         0  unknown  no  
4  unknown    5   may       198         1     -1         0  unknown  no

Las características relevantes que usaremos en este caso son:

balance: saldo bancario; variable clave de interés
job: el título del trabajo ("gerencia", "técnico", etc.)
marital: estado civil ("soltero", "casado" o "divorciado")
education: diferentes niveles de educación ("primaria", "secundaria", "terciaria", "desconocida")
default: el cliente predeterminado ("sí" o "no")
loan: el cliente obtuvo un préstamo ("sí" o "no")

In [5]:
bank.describe()

age        balance           day      duration      campaign  \
count  45211.000000   45211.000000  45211.000000  45211.000000  45211.000000   
mean      40.936210    1362.272058     15.806419    258.163080      2.763841   
std       10.618762    3044.765829      8.322476    257.527812      3.098021   
min       18.000000   -8019.000000      1.000000      0.000000      1.000000   
25%       33.000000      72.000000      8.000000    103.000000      1.000000   
50%       39.000000     448.000000     16.000000    180.000000      2.000000   
75%       48.000000    1428.000000     21.000000    319.000000      3.000000   
max       95.000000  102127.000000     31.000000   4918.000000     63.000000   

              pdays      previous  
count  45211.000000  45211.000000  
mean      40.197828      0.580323  
std      100.128746      2.303441  
min       -1.000000      0.000000  
25%       -1.000000      0.000000  
50%       -1.000000      0.000000  
75%       -1.000000      0.000000  
max      871.000000    275.000000

El año pasado, el saldo promedio de toda la población de clientes del banco fue de \$1341.12. ¿Cuánto más alto es el saldo promedio de la muestra este año? ¿Es esta diferencia significativa?

Respuesta.

In [6]:
bank['balance'].mean()

1362.2720576850766

Esta es una diferencia de $1362.27 - $1341.12 = $22.15. La dirección del banco cree que el comportamiento del consumidor podría haber cambiado ligeramente y que, en promedio, los clientes mantienen más dinero en su cuenta bancaria.

Sin embargo, esta diferencia podría deberse a una variación estadística debido a la variabilidad del muestreo. Necesitaríamos realizar una prueba más rigurosa para determinar si esta diferencia es realmente significativa después de tener en cuenta dicha variabilidad.

### Marco analitico para la prueba de hipotesis

In [10]:
stats.ttest_1samp(bank['balance'], popmean=1341.12) #ese valor es la media con la cual quiero comparar contra el promedio de este a;o
#popmean --> numero con el que quiero comparar el otro arguemnyo (promedio del a;o pasado)
#stats.ttest_1samp(bank['balance'] --> calculame la media de este variable para compararla con la otra

#estadistico: que tanto representa a mi muestra poblacional, 
#estadistico: nos dice que tanta es la diferenica entre la desviacion estandar de una media con respecto a la otra, diferencia entre medias
#el valor 1.47 es bastante chico entonces podemos decue rque son bastante parecidas
#p valor: 0.1396 
#df : cual es la representacion (este no importa)
#ttest_1samp --> este metodo me hace la t student

TtestResult(statistic=1.477137017325908, pvalue=0.13964587808442475, df=45210)

In [8]:
T,p = stats.ttest_1samp(bank['balance'], popmean=1341.122)
p_value = p/2 #lo dividio por dos porque tomo dos anios juntos
p_value

0.06984165666422609

In [11]:
loans=bank[bank.loan=="yes"].balance
no_loans=bank[bank.loan=="no"].balance

statistic, pvalue = stats.ttest_ind(loans,no_loans, equal_var=False)
print('Estadistico:',round(statistic,2),'p-valor:', round(pvalue,2))

Estadistico: -25.18 p-valor: 0.0


Rechazamos la hipotesis nula

Como el p valor es mayor a 0.05 no puedo rechazar la hipotesis nula h0